docs/source/basics/messages.rst

```diff
@@ -0,0 +1,144 @@
+.. _messages_usage_label:
+
+========
+Messages
+========
+
+Messages are a core component in torchtune that govern how text and multimodal content is tokenized. It serves as the common interface
+for all tokenizer and datasets APIs to operate on. Messages contain information about the text content, which role is sending the text
+content, and other information relevant for special tokens in model tokenizers. For more information about the individual parameters
```

Comment:
same comments as before. IMO this should be 1 or 2 lines + maybe image with pipeline + code

Reply:
I personally think this is pretty concise as is. Will add a TODO to add an image. Planning to make an image for the whole data pipeline and I'll place parts of that where relevant in a follow up